src/persist-types/src/lib.rs

```diff
@@ -167,3 +167,21 @@ pub trait Opaque: PartialEq + Clone + Sized + 'static {
     /// have yet been successful.
     fn initial() -> Self;
 }
+
+/// Advance a timestamp by the least amount possible such that
+/// `ts.less_than(ts.step_forward())` is true.
+///
+/// TODO(txn): Unify this with repr's TimestampManipulation. Or, ideally, get
+/// rid of it entirely by making persist-txn methods take an `advance_to`
+/// argument.
+pub trait StepForward {
```

Comment:
I'm not 100% on why we're moving this to a more-shared location, but it certainly doesn't bother me to do so!

Reply:
The impl of this trait will live in mz_repr. Which means that every time I locally invalidate anything in mz-persist-txn during dev (which is constantly), I invalidate basically all of mz. It's certainly awkward, but didn't feel worth it to me to make a whole crate for just this trait when mz-repr already pulls in mz-persist-types